lib/src/config/config.rs

```diff
@@ -119,7 +119,11 @@ impl Config {
                 return Err(self.bad_type(name, val, "an unsigned integer"));
             }
 
-            self.port = port as usize;
+            if port > (u16::max_value() as i64) {
+                return Err(self.bad_type(name, val, "an integer larger than 65535"))
```

Comment:
The string should say what was expected, not what was parsed. In this case, it should say `a 16-bit unsigned integer`.